src/internal/core.rs

```diff
@@ -29,7 +29,7 @@ pub struct State<P: Package, VS: VersionSet, Priority: Ord + Clone> {
 
     /// Store the ids of incompatibilities that are already contradicted
     /// and will stay that way until the next conflict and backtrack is operated.
```

Comment:
Can you improve the comment here to add that we record decision levels to avoid a full reset when backtracking?
That's the only remark I have.

Reply:
Good call. I will fix this tomorrow!